kore/src/Kore/Builtin/AssociativeCommutative.hs

```diff
@@ -123,10 +126,21 @@ class
     -}
     toNormalized
         :: Ord variable
-        => SmtMetadataTools Attribute.Symbol
-        -> TermLike variable
+        => TermLike variable
         -> NormalizedOrBottom normalized variable
 
+    concatNormalized
```

Comment:
`concatNormalized` concatenates two `normalized` which may have become de-normalized, and re-normalizes the result. I wrote it this way because we seem to sometimes need to concatenate things that might have become de-normalized, and it's more efficient to concatenate-and-normalize than to normalize-normalize-concatenate-and-normalize.